### ISSUE TITLE
Post request fails to resolve

### ISSUE THREAD

**Issue by baovitt** [raquo/Laminar#144]
For context, I've been trying to solve this problem on and off for weeks, used several libraries, and every time found a different error. The following code I've written to make a post request to my backend service:
```scala
def fetchJobsFromResume(resume: TapirFile)(using eventBus: EventBus[Event]): EventStream[Event.ResumeEvent] = {
  val encodeRequest = (r: JobSearchWithResumeRequest) => {
    val formData = new org.scalajs.dom.FormData()
    formData.append("file", resume, "experienced-network-engineer-resume-example.pdf")
    formData
  }

  val decodeResponse = (r: org.scalajs.dom.Response) =>
    EventStream.fromJsPromise(r.text()).map(_.fromJson[JobSearchFromResumeResponse])

  FetchStream
    .withCodec(encodeRequest, decodeResponse)
    .post(
      "http://localhost:5000/api/v1/jobs/resume",
      _.headers("accept" -> "application/json"),
      _.body(JobSearchWithResumeRequest(resume))
    ).map(Event.ResumeEvent.UpdateEmbeddingFromResume.apply)
}
```
The call site is a button:
```scala
button(
  cls := "items-center justify-center w-full px-6 py-2.5 text-center text-white duration-200 bg-black border-2 border-black rounded-full nline-flex hover:bg-transparent hover:border-black hover:text-black focus:outline-none lg:w-auto focus-visible:outline-black text-sm focus-visible:ring-black",
  onClick.flatMap { _ => 
    fetchJobsFromResume(resume)
  } --> eventBus.writer,
  "Next"
)
```
The `eventBus` a bus for all application events. I can verify everything is working as expected except the response from the request isn't resolving. The request is made successfully and a response is received successfully as shown by my network tab. Through logging I've found that decodeResponse is never called and the response is never fed into the event bus.

I am using the newest laminar version 16.0.0. I would seriously appreciate any help in solving this. I've spent countless hours trying to fix this and rarely consider reaching out on forums like this. I'm dumbfounded.

**Comment by yurique** [comment]
I'd suggest adding a `.debugLogEvents().debugLogEvents()` to your stream (after `.map(Event.ResumeEvent.UpdateEmbeddingFromResume.apply)` for example) to see if there are any exceptions.

One thing that might or might not be a problem is CORS, but that's just guessing.

**Comment by raquo** [comment]
Hm, this is weird. I can think of three possibilities:

1. Exception, perhaps in your decoder.
2. You are unmounting the button before the request completes, causing the FetchStream to stop due to lack of listeners (because of [laziness](https://github.com/raquo/Airstream/#laziness))
3. CORS is letting you send the request, but is preventing the response from being provided to JS

I suggest:
- Add .setDisplayName(...).debugLifecycle().debugLog() both to the result of `decodeResponse`, and to the result of `fetchJobsFromResume`.
- Check the browser console to see what the log prints, and if there are any errors or exceptions.
- For CORS, check that it's configured properly, including `Access-Control-Allow-Origin`.
- If nothing else helps, try to make the same request using plain JS – for example, have your app dom.console.log the tapir resume, then open dev console, and save that printed object in a variable (google it), then write JS code in the console to call the native JS Fetch with the same url / formdata / other parameters as what you're trying to do here.

If none of this helps, please show the dev console output.

**Comment by baovitt** [comment]
@raquo I had a lot of the same thoughts. 

For cors, everything is running through port 5000. Cors issues I had at first but resolved. I can see the response from the request in the network tab, which shows the request succeeds without any issues. The request is succeeding with status code 200, which also tells me access from the browser isn't an issue.

Even after adding the debugLog to the EventStream, there are no logs in the console, as before. The failure is completely silent.

I'm also certain the request can be made from the browser because I have swagger docs which let me use the endpoint. Beside that, I can see the successful response body in the network tab. The request is succeeding with status code 200.

I also tried modifying the decoder with a log statement:
```scala
val decodeResponse = (r: org.scalajs.dom.Response) =>
      println("decodeResponse")
      EventStream.fromJsPromise(r.text()).map(_.fromJson[JobSearchFromResumeResponse])
```
"decodeResponse" never prints to the console, even after the request succeeds.

In summary, the request is clearly succeeding, there are no error logs in the console, and "decodeResponse" isn't running.

One note: the other day I tried having the request run when the button mounted, instead of when it gets clicked. The button remounted about 100 times every second causing about 700 requests to my backend. Do you suppose this could be the issue?

For greater context, the can inspect the code [here](https://github.com/baovitt/jobgun/blob/main/modules/frontend/src/main/scala/com/jobgun/ui/components/tryfree/ResumeForm.scala):

**Comment by yurique** [comment]
So you're rendering the page inside this: 
`child <-- globalState.map {`

When the button is clicked, it emits an event into the event bus and that, in turn, causes the `globalState` signal to update, which will cause the button to re-render (removing the old button from the DOM and cancelling all subscriptions). 

If I'm reading it right, this looks like the Option 2 Nikita mentioned above.

**Comment by raquo** [comment]
What Yurique said. Laminar has no virtual dom – this has many advantages, but needs appropriate techniques when rendering stateful components with `child <--` or `children <--`.

See:
https://www.youtube.com/watch?v=L_AHCkl6L-Q&t=767s
https://laminar.dev/documentation#lists-of-children
https://laminar.dev/documentation#performant-children-rendering--split
https://laminar.dev/documentation#splitting-individual-items

These talk mostly about rendering lists of children, but the issues that they explain apply to rendering individual children (like in your case) just as well.

**Comment by baovitt** [comment]
Thank you! I can finally make progress again.
